File: pubnub-java/src/main/java/com/pubnub/internal/endpoints/push/ListPushProvisionsImpl.java
package com.pubnub.internal.endpoints.push;

import com.pubnub.api.enums.PNPushEnvironment;
import com.pubnub.api.enums.PNPushType;
import com.pubnub.api.models.consumer.push.PNPushListProvisionsResult;
import com.pubnub.internal.InternalPubNubClient;
import com.pubnub.internal.endpoints.DelegatingEndpoint;
import lombok.Setter;
import lombok.experimental.Accessors;

@Setter
@Accessors(chain = true, fluent = true)
public class ListPushProvisionsImpl extends DelegatingEndpoint<PNPushListProvisionsResult> implements com.pubnub.api.endpoints.push.ListPushProvisions {

    private PNPushType pushType;
    private String deviceId;
    private PNPushEnvironment environment;
    private String topic;

    public ListPushProvisionsImpl(InternalPubNubClient pubnub) {
        super(pubnub);
    }

    @Override
    protected ListPushProvisions createAction() {
        return pubnub.auditPushChannelProvisions(
                pushType,
                deviceId,
                topic,
                environment
        );
    }
}

File: pubnub-kotlin/src/main/kotlin/com/pubnub/internal/endpoints/push/ListPushProvisionsImpl.kt
package com.pubnub.internal.endpoints.push

import com.pubnub.api.endpoints.push.ListPushProvisions
import com.pubnub.internal.PubNubImpl

/**
 * @see [PubNubImpl.auditPushChannelProvisions]
 */
class ListPushProvisionsImpl internal constructor(listPushProvisions: ListPushProvisionsInterface) :
    ListPushProvisionsInterface by listPushProvisions, ListPushProvisions
